subtests/docker_cli/run_cgroups/cgroup_parent.py

```diff
@@ -0,0 +1,103 @@
+#. Compare to expected result
+"""
+
+
+import os.path
+import re
+from autotest.client import utils
+from dockertest.dockercmd import DockerCmd
+from dockertest.images import DockerImage
+from dockertest.output import OutputGood
+from dockertest.xceptions import DockerTestError, DockerTestFail
+from cgroups_base import cgroups_base
+
+
+class cgroup_parent_base(cgroups_base):
+    def _setup(self, cg_parent, expected_path, expected_stderr):
+        """
+        Set up test parameters. Our caller gives us a value to use
+        for --cgroup-parent, and what to expect in stdout & stderr.
+        We replace the format strings {rand1} and {rand2} with two
+        pseudorandom strings, and {cid} with the container ID (once
+        we have it).
+        """
+        self.sub_stuff['rand1'] = utils.generate_random_string(8)
+        self.sub_stuff['rand2'] = utils.generate_random_string(8)
+        self.sub_stuff['cgroup_parent'] = cg_parent.format(**self.sub_stuff)
+        self.sub_stuff['expected_path'] = expected_path
+        self.sub_stuff['expected_stderr'] = expected_stderr
+
+    def run_once(self):
+        """
+        Run docker with the given --cgroup-parent; preserve cgroup info.
+        """
+        super(cgroup_parent_base, self).run_once()
+
+        cidfile = os.path.join(self.tmpdir, 'cidfile')
+        self.sub_stuff['cidfile'] = cidfile
+
+        subargs = ['--rm',
+                   '--cgroup-parent=%s' % self.sub_stuff['cgroup_parent'],
+                   '--cidfile', cidfile,
+                   DockerImage.full_name_from_defaults(self.config),
+                   '/bin/cat', '/proc/1/cgroup']
+        dc = DockerCmd(self, "run", subargs,
+                       timeout=self.config['docker_timeout'])
+        self.sub_stuff["cmdresult"] = dc.execute()
+
+    def postprocess(self):
+        super(cgroup_parent_base, self).postprocess()
+
+        cmdresult = self.sub_stuff["cmdresult"]
+        OutputGood(cmdresult, ignore_error=False)
+
+        # Expected output may include the container ID; allow caller
+        # to use "{cid}", which we replace with actual one for this run.
+        cid = open(self.sub_stuff['cidfile'], 'rb').read().strip()
+        if len(cid) < 12:
+            raise DockerTestError("bad cid (length < 12) in --cidfile")
+        self.sub_stuff['cid'] = cid
+        path_exp = self.sub_stuff['expected_path'].format(**self.sub_stuff)
+        stderr_exp = self.sub_stuff['expected_stderr'].format(**self.sub_stuff)
+
+        # Check stderr first: that way if we're not expecting an error, but
+        # get one, our user has a better chance of seeing something useful.
+        # (Note: Running docker with -D (debug) produces lots of cruft:
+        #    time="...." level=debug msg="unwanted stuff"
+        # Strip them out.)
+        stderr = "\n".join([line
+                            for line in cmdresult.stderr.strip().split("\n")
+                            if not line.startswith('time="')])
```

Comment:
Just wait and see, they'll change the format in the next version, then three more times after that..  Feel free to just examine ``self.config['docker_options']`` and strip out any '-D' since it impacts the test so greatly.  TBH, I'm not sure I even like the '-D' in there by default.  It's not really ever been that useful, even when things do go south.  I'd also support nuking it from ``defaults.ini``, but I suppose you never know when it might get switched back on.  Uggg, it's never easy.

Maybe stderr only needs to be examined if the exit code is non-zero?

Hmmm.

Failing to think how this could be made simpler / avoided entirely.